parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreV2.java

```diff
@@ -147,6 +148,11 @@ public void endRecord() {
   }
 
   private void sizeCheck() {
+    if(!parquetProperties.isEstimateNextSizeCheck()) {
```

Comment:
This block only sets the next row count for a size check and leaves out checking whether the pages should be written.